libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/router/NavigationRoute.kt

```diff
@@ -0,0 +1,798 @@
+package com.mapbox.navigation.route.offboard.router
+
+import android.content.Context
+import androidx.annotation.FloatRange
+import androidx.annotation.IntRange
+import com.mapbox.api.directions.v5.DirectionsCriteria
+import com.mapbox.api.directions.v5.MapboxDirections
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.core.exceptions.ServicesException
+import com.mapbox.core.utils.TextUtils
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.route.offboard.extension.getUnitTypeForLocale
+import com.mapbox.navigation.utils.extensions.inferDeviceLocale
+import okhttp3.EventListener
+import okhttp3.Interceptor
+import retrofit2.Call
+import retrofit2.Callback
+
+/**
+ * The NavigationRoute class wraps the [MapboxDirections] class with parameters which
+ * <u>must</u> be set in order for a navigation session to successfully begin. While it is possible
+ * to pass in any [com.mapbox.api.directions.v5.models.DirectionsRoute] into
+ * [MapboxNavigation.startNavigation], using this class will ensure your
+ * request includes all the proper information needed for the navigation session to begin.
+ *
+ * Developer Note: MapboxDirections cannot be directly extended since it is an AutoValue class.
+ *
+ * 1.0
+ */
+
+internal class NavigationRoute
```

Comment:
Look at comment above 👀 

Right now we're not taking `Route` / `RouteOptionsNavigation` (`NavigationRoute`) as a parameter so none of the options from `NavigationRoute` can be configured from a client point of view. Should we request a `Route` / `RouteOptionsNavigation` (`NavigationRoute`) object here (that includes `origin`, `waypoints`, `destination` + other fields) instead of `origin`, `waypoints` and `destination`? That was the purpose of having `NavigationRoute` and we're not using it, we're only setting these 3 parameters and not considering the rest.

Shouldn't the builder for configuring the different options live in `RouteOptionsNavigation` (`base` module) and removed from here?